### PR TITLE
Created coverage reporter for jest-expo

### LINKED ISSUE #4644: [expotools] Generate and upload code coverage reports
`check-packages` command from `expotools` can also upload code coverage information to Codecov. Codecov supports "flags" for each coverage report. Flags let us annotate a report for this repo with the name of the unimodule we tested. This way for one repo we can have multiple code coverage reports. https://docs.codecov.io/docs/flags

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

# How

Collects coverage for each test individually accepts `universal` or `enzyme`.
Coverage is output in platform specific folders `coverage/{platform}`. 
By default each reporter can only ignore non-matching platform extensions, for instance web won't include `.ios`, `.android`, `.node`, or `.native` files. Unfortunately this means if you have a file like `foo.ts` and `foo.web.ts`, `foo.ts` will show as having 0% coverage.

Rules regarding `Platform.OS` and `Platform.select` are also not respected by Jest's coverage reporter.

# Test Plan

- coverage should be generated
- coverage folder is git ignored and npm ignored

### CODE REVIEW

File: packages/expo-module-scripts/templates/.npmignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 # @generated by expo-module-scripts
 babel.config.js
+/coverage
```
Comment (ide) [inline packages/expo-module-scripts/templates/.npmignore]:
```suggestion
/coverage/
```

File: packages/expo-module-scripts/bin/expo-module-coverage [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+set -eo pipefail
+
+script_dir="$(dirname "$0")"
+
+"$script_dir/npx" jest --coverage --preset expo-module-scripts/"$1"/ios
+"$script_dir/npx" jest --coverage --preset expo-module-scripts/"$1"/android
+"$script_dir/npx" jest --coverage --preset expo-module-scripts/"$1"/web
+
+if [[ $1 = 'universal' ]]; then
+    "$script_dir/npx" jest --coverage --preset expo-module-scripts/"$1"/node
```
Comment (ide) [inline packages/expo-module-scripts/bin/expo-module-coverage]:
```suggestion
  "$script_dir/npx" jest --coverage --preset expo-module-scripts/"$1"/node
```

**Review by ide — APPROVED** [review]

**Comment by tsapeta** [comment]
FYI: I've updated PR description to link it with #4644 and automatically close when merged.

**Comment by EvanBacon** [comment]
closing, stale

**Pull request closed without merging**